server.go

```diff
@@ -195,18 +195,20 @@ func WithCorsMiddleware(corsMiddleware func(http.Handler) http.Handler) func(*Se
 }
 
 // WithGlobalResponseTypes adds default response types to the server.
-// Useful for adding global error types.
 // For example:
 //
 //	app := fuego.NewServer(
 //		fuego.WithGlobalResponseTypes(400, "Bad Request _(validation or deserialization error)_", HTTPError{}),
 //		fuego.WithGlobalResponseTypes(401, "Unauthorized _(authentication error)_", HTTPError{}),
 //		fuego.WithGlobalResponseTypes(500, "Internal Server Error _(panics)_", HTTPError{}),
+//		fuego.WithGlobalResponseTypes(204, "No Content", Empty{}),
 //	)
-func WithGlobalResponseTypes(code int, description string, errorType ...any) func(*Server) {
-	errorType = append(errorType, HTTPError{})
+func WithGlobalResponseTypes(code int, description string, response Response) func(*Server) {
 	return func(c *Server) {
-		c.OpenAPI.globalOpenAPIResponses = append(c.OpenAPI.globalOpenAPIResponses, openAPIError{code, description, errorType[0]})
+		c.OpenAPI.globalOpenAPIResponses = append(
+			c.OpenAPI.globalOpenAPIResponses,
+			openAPIResponse{Code: code, Description: description, Response: response},
+		)
 	}
 }
 
```

Comment:
Now it is the same as this :

```go
func WithGlobalResponseTypes(code int, description string, response Response) func(*Server) {
	return func(c *Server) {
		WithRouteOptions(
			OptionAddResponse(code, description, response),
		)(c)
	}
}
```


Or, can we remove it completely? I think it's better if we remove redundant configuration items. I think that everything that will be applied to a route must be applied as a route option.

Reply:
Hmm let met test this real quick. 

I personally really like being allowed to set global response types for servers/groups.

Reply:
Yeah these two things aren't the same.

```
--- FAIL: TestWithGlobalResponseType (0.00s)
    --- FAIL: TestWithGlobalResponseType/errors_with_route_overrides (0.00s)
        openapi_operations_test.go:104: 
                Error Trace:    /Users/dhitt/github/go-fuego/fuego/openapi_operations_test.go:104
                Error:          Not equal: 
                                expected: "My Global Error"
                                actual  : "Bad Request _(validation or deserialization error)_"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -My Global Error
                                +Bad Request _(validation or deserialization error)_
                Test:           TestWithGlobalResponseType/errors_with_route_overrides
                Messages:       Overrides Fuego's default 400 error
```

Reply:
I guess we can just use WithRouteOptions here as well. I forget that exists always. Let me figure out the reason for the error above and I'll submit another PR

Reply:
At least I tested this well enough 😅

Reply:
Okay, nice catch. Just needed to change up this: https://github.com/go-fuego/fuego/pull/276/commits/81a7afcda321dc40268a9f1b0d08a7d7d7dc2feb

So couple things. Yes we should deprecate `WithGlobalResponseType`. I think we should get this change merge with a deprecation notice and best way to migrate to `WithRouteOptions`. Emphasis on maintaining the two default api responses declare as well. I'll probably move those over to use `WithRouteOptions` in this MR as well so we don't forget. 

How does that sound?

Reply:
I think it looks good!

And yes moving back the default server options is really nice as well, that's something we wanted to do anyway.

Reply:
Okay cool, I’ll get this don’t either tonight or tomorrow night then.